src/aarch64.rs

```diff
@@ -0,0 +1,808 @@
+            // intrinsics being used. Sizes are compile-time checked in test builds.
+            unsafe { arch::$intrinsic(::core::ptr::from_mut(into).cast(), from) }
+        }
+    };
+}
+
+macro_rules! assert_size_8bytes {
+    ($n:literal registers $ty:ty as $real:ty) => {
+        // Ensure that the type has individual vectors 64 bits wide.
+        const _: () = ::core::assert!(::core::mem::size_of::<$ty>() == 8 * $n);
+        const _: () = ::core::assert!(::core::mem::size_of::<$real>() == 8 * $n);
+    };
+}
+
+macro_rules! assert_size_16bytes {
+    ($n:literal registers $ty:ty as $real:ty) => {
+        // Ensure that the type has individual vectors 128 bits wide.
+        const _: () = ::core::assert!(::core::mem::size_of::<$ty>() == 16 * $n);
+        const _: () = ::core::assert!(::core::mem::size_of::<$real>() == 16 * $n);
+    };
+}
+
+#[cfg(test)]
+macro_rules! various_sizes {
+    ($n:literal registers $ty:ty as $real:ty) => {
+        // Only make sure that our annotated structure and the interface type match up
+        const _: () = ::core::assert!(core::mem::size_of::<$ty>() == core::mem::size_of::<$real>());
+    };
+}
+
+#[cfg(not(test))]
+macro_rules! various_sizes {
+    // No reason to take up compile time for a constant assert.
+    ($n:literal registers $ty:ty as $real:ty) => {};
+}
+
+// There are four fundamental types of loads:
+// - `vldN[q]_<ty>` which loads an array of structures of N elements of type <ty>, as many as
+//   fill the 8-byte or with q 16-byte registers. Eg. vld2q_f32 would load 8 total values, each
+//   of the two result vectors getting 4 elements for a total of 16-bytes each. (These loads
+//   would be interleaved but that's not important for the memory access part).
+//
+// - `vld1[q]_<ty>_xK` which is performs multiple loads of the type `vld1[q]_<ty>` to
+//   K consective registers but in a single instruction, compared to stacking multiple loads.
+//
+// - vldN[q]_dup_<ty> which loads *one* structure of N elements and fills each of the N
+//   registers with the contents of its one element.
+//
+//- `vldN[q]_lane_<t> which loads *one* structure of N elements and inserts the contents of
+//   each registers elements into a specific lane, i.e. the lane must be within the bounds such
+//   that `[ty; LANE]` does not exceed 8 or 16-bytes with/without q respectively.
+vld_n_replicate_k! {
+    unsafe: load;
+    // Loads full registers, so 8 bytes per register
+    size: assert_size_8bytes;
+
+    /// Load an array of 8 `u8` values to one 8-byte register.
+    fn vld1_u8(_: &[u8; 8][..1] as [u8; 8]) -> uint8x8_t;
+    /// Load an array of 8 `i8` values to one 8-byte register.
+    fn vld1_s8(_: &[i8; 8][..1] as [i8; 8]) -> int8x8_t;
+    /// Load an array of 4 `i16` values to one 8-byte register.
+    fn vld1_u16(_: &[u16; 4][..1] as [u16; 4]) -> uint16x4_t;
+    /// Load an array of 4 `u16` values to one 8-byte register.
+    fn vld1_s16(_: &[i16; 4][..1] as [i16; 4]) -> int16x4_t;
```

Comment:
`i16`/`u16` doc comments are swapped here still.

It looks like you got all the others